src/main/java/com/atlassian/db/replica/api/aurora/AuroraReplicasDiscoverer.java

```diff
@@ -0,0 +1,55 @@
+package com.atlassian.db.replica.api.aurora;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static com.atlassian.db.replica.api.aurora.AuroraEndpoints.replicaUrl;
+
+/**
+ * It allows to pull Aurora Replicas cluster information
+ */
+public class AuroraReplicasDiscoverer {
+    private final Connection connection;
+
+    public AuroraReplicasDiscoverer(Connection connection) {
+        this.connection = connection;
+    }
+
+    /**
+     * It provides list of replica urls based on reader endpoint and server ids
+     * @param readerEndpoint - Aurora reader endpoint
+     * @return list of replicas jdbcUrls
+     * @throws SQLException
+     */
+    public List<String> fetchReplicasUrls(String readerEndpoint) throws SQLException {
+        return fetchReplicaServerIds()
+            .stream()
+            .map(serverId -> replicaUrl(
+                readerEndpoint,
+                serverId
+            )).collect(Collectors.toList());
+    }
+
+    /**
+     * Asks Aurora about the list of server ids
```

Comment:
It seems like a behavior of `AuroraCluster`. It could ask for the `Connection` via method param.

Reply:
It depends, I was looking at them as pure data classes. They simply transfer data and are not coupled with behavior.
However, I'm biased here so happy to consider pros and cons.

Reply:
Pure data classes = procedural style

Reply:
Not sure, these are encouraged everywhere lately, even Java is adding them "natively" (Java Records).

Reply:
Records are allowed to have methods/behaviors. Who's encouraging anemic DTO classes?

Reply:
What's the advantage of doing this? Currently, AuroraCluster is just storing data and is immutable, easy to reason about.
Once you start adding more methods you have watch-out for internal state, mutability etc.

Not to mention you have to start adding annotations like `@JsonIgnore`, because you never know...

My thinking right now: _I don't need AuroraCluster in order to fetch server ids. It's already 100lines class, throwing more stuff in there won't make it more readable_

Reply:
> Not to mention you have to start adding annotations like `@JsonIgnore`. 

wat

Reply:
I can move it to `AuroraCluster`, but shouldn't we move `fetchReplicasEndpoints` as well?

Anyway main pro is see: extracting `fetchReplicasServerIds` to a different place will help us to mock this dependency.

Reply:
Yes, we should move both fetches.
You can mock an entire `AuroraCluster`.